### DISCUSSION TITLE
Reconsider dashboard default in config:base

### DISCUSSION THREAD

**Discussion by rarkins** [renovatebot/renovate#11415] [Suggest an Idea]
Some people who manage dozens or hundreds of open source repositories using Renovate were surprised to see Dashboard issues created today when the app went live with v26. It has since been rolled back pending this discussion.

Should we disable dashboard by default?

I think the majority of users would benefit from it being enabled by default, but the question is how to handle the minority and whether it is an unwelcome change we should avoid. In general we should follow the [principle of least surprise](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) which would mean no dashboard unless opted into.

@viceice @JamieMagee @HonkingGoose

**Comment by rarkins** [comment]
The app has been rolled back to v25. Any dashboards created will not be cleaned up until we implement #11414, which we should do before updating to v26 again regardless.

If we decide to keep dashboards, do we have a way of "warning" people using v25 via the app about it? And can we wait that long anyway to keep holding back v26?

If we disable default dashboards again, what to do about people using v26? Should we bump to v27 for them out of courtesy?

**Reply by viceice** [reply]
Yes we should add #11414, can do this in a few minutes

**Reply by HonkingGoose** [reply]
> If we disable default dashboards again, what to do about people using v26? Should we bump to v27 for them out of courtesy?

If we as a project decide to rollback on the idea of "Use dashboard by default, we should bump to `v27`" as that denotes a clear break in the behavior between versions. I don't think switching our defaults yet again is something we should do though.

**Comment by viceice** [comment]
I don't like to disable the dashborad again, as i think it will be more helpful to be enabled by default. 

I know some people are suprised from it, they mostly tend to cry about it, because they don't know how to use it.

I think there are only a few people who really want to have the dashboard disabled after some thoughts about it.

Can we add the disable dashboard preset temporary to the global app config ? will it disable the dashboard again for the app?

Aren't most dashboard issues now created by the app? so we would make even more noise to close them all again?

We should / can use the dashboard footer to add some more infos / links about the dashboard.

**Reply by rarkins** [reply]
Not too many were created yet by the app as it ran for a short time.

If we disable it in the app then it might disable people's intentionally enabled dashboards too.

**Reply by rarkins** [reply]
Maybe roll back in v27, add a note to all PRs with a link to a blog post or something, and then enable in v28?

**Reply by viceice** [reply]
😕 Would cause a lot config changes again for me, i've already migrated to v26

**Reply by HonkingGoose** [reply]
## On dependency dashboard default behavior

> I don't like to disable the dashboard again, as i think it will be more helpful to be enabled by default.

Agree, I think having the dashboard should be a net-positive for most users. It gives a overview of the status of the repository, and shows all dependencies yet to be updated. I don't think we should back down on this decision just yet.

> I know some people are surprised from it, they mostly tend to cry about it, because they don't know how to use it.

Big changes like this can be surprising/scary. So we should try to explain our reasoning, and direct users to an easy way to "opt-out" again. We could direct users to our new `key-concepts/dashboard.md` page to help educate them maybe?

## On using dependency dashboard footer

> We should / can use the dashboard footer to add some more infos / links about the dashboard.

I had this idea as welll, but @rarkins didn't like that (see https://github.com/renovatebot/renovate/discussions/11170):

@HonkingGoose said: 

> If we decide to turn it on by default in `v26`, maybe we should add a new footer text to the dashboard:
> 
> ```gfm
> If you don't want this Dependency Dashboard, do .... to disable it.
> ```
> 
> This way people don't have to go searching in our docs for how to turn off the dashboard.
> It might be a bit of a "surprise" for people who get automatically updated (GitHub app users), so that's why I want to provide an easy way to learn how to turn it off again. 😄

@rarkins responded:

> I don't consider this strictly to be a breaking change (it's very hard to define breaking when things come to user interface) but adding to v26 now is a good idea so there's no need to debate it. I prefer not to permanently pollute the dashboard with "how to turn it off" though, so maybe we just need to make sure that doing so is easily searchable in our docs? Especially near the top of the dependency dashboard page/section.

But maybe the situation and opinion has changed now, and we value explaining the new behavior and provide a way to opt-out higher than having a clean dashboard? We could also put the new footer in `v26` and remove it in `v27` or some later version. So we don't _permanently_ clutter up the dashboard?

**Comment by viceice** [comment]
One other issue is, the docs are already on v26, so we get more and more issues about not working stuff, because of wrong configs

**Reply by HonkingGoose** [reply]
That's another thing to keep in mind! Rolling back the changes on the docs is hard, because I already wrote the new docs with `v26` behavior in mind! Right now the docs and the program version are coupled together.

**Comment by rarkins** [comment]
@JustinBeckwith the consensus here and including @JamieMagee in a Slack conversation is to keep the dashboard as enabled by default if using the `config:base` preset. I do think you'd benefit from some visibility into PRs in your repos, but also happy to add a server side rule if you prefer. Can you confirm you want dashboards force disabled across all googleapis repos, and any others? I don't have visibility into whether any of the repos have *enabled* them manually until now.

**Reply by JustinBeckwith** [reply]
Thanks for jumping on this so quickly folks - I really appreciate you taking a step back to walk through things :) On force disabling - I don't know that we need any special treatment here.  It was just a surprise to see hundreds of issues to start flying by 😆  I am going to pull in @bcoe here for his opinion, because his team will likely be the ones dealing with any fallout here.

**Reply by rarkins** [reply]
If you'd like to see the dashboard in action,  check out the pinned issue in this repo

**Reply by rarkins** [reply]
@bcoe any last requests before we roll ahead a second time?

**Reply by bcoe** [reply]
No requests. I think for our specific workflow on Node.js, we may turn off this feature. But, we can do this through our extensions to the config.

**Comment by mondeja** [comment]
I've considered this feature as spam because I've needed to create 40 commits in 40 repositories. What are you thinking about? If it is enabled by default, why I should enable it explicitly, according to the documentation? Also, some developers are now worrying if normal updates are still active if we disable Dependency Dashboard, and now I'm worrying about the same. Could you, at least, point that normal updates will still be enabled even disabling this feature?

And please, don't create new features spamming in all repositories :pray:

**Reply by rarkins** [reply]
You already edited your comment 5 times. How about one more edit to make it genuinely polite? 

For example: remove inflammatory terms like "spam" and "spamming" and rude sentences like "What are you thinking about?" when this Discussion literally shows our thoughts.

**Reply by HonkingGoose** [reply]
> I've considered this feature as spam because I've needed to create 40 commits in 40 repositories.

I think you mean that you needed 40 commits to adjust the renovate config file so that the dashboard is turned off again?

> What are you thinking about?

To clarify, we're thinking about changing our `config:base` preset, that people will get by default from our Renovate onboarding config PR.
The `config:base` preset for `v25` says: "Dashboard disabled by default".

We intend for `v26`'s `config:base` preset to have the default of "dashboard turned on, unless disabled by end-users". I believe that many people will benefit from the dashboard.

> If it is enabled by default, why I should enable it explicitly, according to the documentation?

If you _do not_ `extend` from `config:base`, you need to enable the dashboard manually. If you _do_ extend from `config:base` the new default is that the dashboard is turned on, so you won't need to enable it explicitly.

> Also, some developers are now worrying if normal updates are still active if we disable Dependency Dashboard, and now I'm worrying about the same. Could you, at least, point that normal updates will still be enabled even disabling this feature?

Whatever happens with the dashboard will not affect any updates from Renovate PRs. You'll still get Renovate update PRs as normal.

If you do not want the dashboard you can put this in your `renovate.json` config file and turn off the dashboard for that repository:

```json
{
  "extends": ["config:base", ":disableDependencyDashboard"]
}
```

**Reply by mondeja** [reply]
> I think you mean that you needed 40 commits to adjust the renovate config file so that the dashboard is turned off again?

Yes, in a year possibly will be 60.

>  I believe that many people will benefit from the dashboard.

Based on what? If you think that could benefit, you will force to a lot of users to do a lot of commits just to disable it? Why you've though that making a disabled by default feature could benefit making it enabled by default when they can't control the version used by Renovate? I don't understand it. I should suppose that the reason is "I believe that many people will benefit from the dashboard"? And do you've an objetive reason or just your feelings?

**Reply by viceice** [reply]
@mondeja If you have 40 repo's, you should consider using a custom shared preset

https://docs.renovatebot.com/config-presets/

**Reply by rarkins** [reply]
I am locking this discussion due to the impolite tone being used and refusal to amend it, plus we already considered and decided to keep the dashboard so the decision is done.

For the record:
- Dependency Dashboard is an optional feature, but we consider it to be a core part of Renovate we recommend
- Dependency Dashboard is disabled by default by Renovate, but enabled in the `config:base` preset which is the default onboarding config for the hosted app
- Users are not forced to use `config:base`, and can stop using it or override it (you do not need to "control the version used by Renovate")
- We recommend that people with 10s of repos or more use their own config preset and then it's just one commit to change settings, not 40 or 60.
- You can then automatically get such a personal preset used as default onboarding config by putting a file like this in their `.github` org/user repo: https://github.com/renovatebot/.github/blob/main/renovate-config.json

@mondeja you can also uninstall the app and self-host, where the `config:base` is not even the default onboarding config. 

But if you post here again, please go to efforts to be fully polite. Imagine if people jumped into your repos insisting with rudeness that you weren't allowed to make decisions on your own products even when you "believe that many people will benefit from X".

**Reply by rarkins** [reply]
One more point: if anyone reading this wants me to try putting a server side rule to ignore the dashboard preset for your org/account, please start a new "App Support" discussion and I can add it manually for you.
